sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.netstandard2.0.cs

```diff
@@ -5,26 +5,32 @@ namespace Azure.Core
     {
         private readonly object _dummy;
         private readonly int _dummyPrimitive;
-        public BinaryData(object data, System.Type type, Azure.Core.Serialization.ObjectSerializer? serializer = null) { throw null; }
+        public BinaryData(byte[] data) { throw null; }
+        public BinaryData(object serializable, Azure.Core.Serialization.ObjectSerializer serializer, System.Type? type = null) { throw null; }
+        public BinaryData(object jsonSerializable, System.Type? type = null) { throw null; }
```

Comment:
ugh

Reply:
This is also unsafe. JsonSerializer always takes type for a reason, sometimes a subclass contains more information than the type you were intending to serialize, not having to pass type might result in a security issue.

Reply:
/cc @KrzysztofCwalina @tg-msft

Reply:
This was Krzysztof's suggestion.  The feeling was this is a customer API that they call to get data so it's okay to leave the choice up to them while defaulting to the current instance's type.